internal/provider/kubernetes/predicates.go

```diff
@@ -181,6 +181,28 @@ func (r *gatewayAPIReconciler) validateSecretForReconcile(secret *corev1.Secret)
 		}
 	}
 
+	if r.backendCRDExists {
```

Comment:
Does the same problem exist for configmap/clustertrustebundle?

Reply:
It's already handled here?

https://github.com/envoyproxy/gateway/blob/650fd83faac89ee5f0af59c8653ee032431e625a/internal/provider/kubernetes/predicates.go#L187-L207

Reply:
looks like we missed for ConfigMap?